lib/crowbar/client/command/proposal.rb

```diff
@@ -101,33 +104,114 @@ module Proposal
             parent.arg :barclamp
             parent.arg :proposal
-            parent.command :create do |c|
+            parent.command "create" do |c|
+              c.desc "Proposal data in json format"
+              c.flag [:data], type: String, default_value: nil
+
+              c.desc "Proposal data as a json file"
+              c.flag [:file], type: String, default_value: nil
+
               c.action do |global, opts, args|
                 barclamp = args.shift
                 proposal = args.shift
                 helper.validate_availability_of! barclamp
 
-                raise "Not implemented yet! (#{barclamp}, #{proposal})"
+                # TODO(should): Preload the template from API
+
+                content = case
+                when opts[:data]
+                  begin
+                    JSON.load opts[:data]
+                  rescue
+                    exit_now! "Invalid json data"
+                  end
+                when opts[:file]
+                  begin
+                    JSON.load opts[:file]
+                  rescue
+                    exit_now! "Invalid json file"
+                  end
+                else
+                  begin
+                    editor = Util::Editor.new content: {}
+                    editor.edit!
+
+                    editor.result
+                  rescue EditorAbortError => e
+                    exit_now! e.message
+                  rescue EditorStartupError => e
+                    exit_now! e.message
+                  rescue InvalidJsonError => e
+                    exit_now! e.message
+                  rescue
+                    exit_now! "Editing content failed"
+                  end
+                end
+
+                raise content.inspect
+
               end
             end
 
             parent.desc "Edit a proposal for specific barclamp"
             parent.arg :barclamp
             parent.arg :proposal
-            parent.command :edit do |c|
+            parent.command "edit" do |c|
+              c.desc "Proposal data in json format"
+              c.flag [:data], type: String, default_value: nil
+
+              c.desc "Proposal data as a json file"
+              c.flag [:file], type: String, default_value: nil
+
               c.action do |global, opts, args|
                 barclamp = args.shift
                 proposal = args.shift
                 helper.validate_availability_of! barclamp
 
-                raise "Not implemented yet! (#{barclamp}, #{proposal})"
+                original = Request.instance.proposal_show(
+                  barclamp,
+                  proposal
+                ).parsed_response
+
+                content = case
+                when opts[:data]
+                  begin
+                    JSON.load opts[:data]
+                  rescue
+                    exit_now! "Invalid json data"
+                  end
+                when opts[:file]
```

Comment:
Style/CaseIndentation: Indent `when` as deep as `case`. (https://github.com/bbatsov/ruby-style-guide#indent-when-to-case)